### PR TITLE
Rbsfxirjtp 233 aurora service unit tests

### PR DESCRIPTION
**Pull request by nikolanguen**

### CODE REVIEW

File: aurora/src/test/java/com/market/banica/aurora/config/ChannelManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    private static final String CHANNEL_PREFIX_ASTERISK = "*";
+
+    private static final String HOST = "localhost";
+    private static final int PORT = 1010;
+    private static final int EDITED_PORT = 1011;
+    private static ChannelProperty channelProperty;
+
+
+    @Mock
+    private ManagedChannel managedChannel = ManagedChannelBuilder
```
Comment (Lilyanov) [inline aurora/src/test/java/com/market/banica/aurora/config/ChannelManagerTest.java:37]: why is this mocked and initialized at the same time. Shouldn't it be mocked only ?

File: aurora/src/test/java/com/market/banica/aurora/config/JMXConfigTest.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+    private static final String EDITED_HOST = "edited-localhost";
+
+    private static final String TEST_FILE_NAME = "test";
+
+    @Mock
+    private ChannelManager channels;
+
+    @Spy
+    private final Map<String, ChannelProperty> channelPropertyMap = new HashMap<>();
```
Comment (Lilyanov) [inline aurora/src/test/java/com/market/banica/aurora/config/JMXConfigTest.java:51]: why is this @spy and initialized at the same time, I think @Spy needs to be removed

File: aurora/src/test/java/com/market/banica/aurora/handlers/RequestHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class RequestHandlerTest {
+
```
Comment (Lilyanov) [inline aurora/src/test/java/com/market/banica/aurora/handlers/RequestHandlerTest.java:31]: This should be removed

File: aurora/src/test/java/com/market/banica/aurora/AuroraApplicationTest.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package com.market.banica.aurora;
+
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+
+@SpringBootTest
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_CLASS)
+public class AuroraApplicationTest {
```
Comment (trayan-muchev) [inline aurora/src/test/java/com/market/banica/aurora/AuroraApplicationTest.java:10]: This class is empty at the momment why do we need it?

File: aurora/src/test/java/com/market/banica/aurora/config/ChannelManagerTest.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        List<ManagedChannel> expectedList = Arrays.asList(marketWeather, marketChannel);
+        //Act
+        List<ManagedChannel> actualList = channelManager.getAllChannelsContainingPrefix(CHANNEL_PREFIX_ASTERISK);
+
+        //Assert
+        assertEquals(expectedList, actualList);
+    }
+
+    @Test
+    void addChannelAddsChannelInMap() {
```
Comment (trayan-muchev) [inline aurora/src/test/java/com/market/banica/aurora/config/ChannelManagerTest.java:69]: Test names except for the first test getAllChannelsContainingPrefixAsteriskReturnsAllChannels are not clear enough as there is no condition in them.
Suggestion: addChannelWithValidInputSuccessfullyAddedChannelInMap
This applies for all test names below.

File: aurora/src/test/java/com/market/banica/aurora/config/JMXConfigTest.java [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+    }
+
+    @Test
+    void getChannelsStatusVerifiesGetChannelsMethodCall() {
+        jmxConfig.getChannelsStatus();
+        Mockito.verify(channels, Mockito.times(1)).getChannels();
+    }
+
+    @Test
+    void writeBackUp() {
```
Comment (trayan-muchev) [inline aurora/src/test/java/com/market/banica/aurora/config/JMXConfigTest.java:149]: Test name is not clear enough as there is no condition and expected result.

File: aurora/src/test/java/com/market/banica/aurora/handlers/SubscribeHandlerTest.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+        //Act
+        subscribeHandler.handleSubscribe(AURORA_REQUEST_BANICA, auroraResponse);
+
+        //Assert
+        verify(auroraResponse, times(1)).onError(any());
+    }
+
+//    @Test
+//    void handleSubscribeWithAuroraRequestForExistingChannelInvokesStubSubscribeMethod() {
```
Comment (trayan-muchev) [inline aurora/src/test/java/com/market/banica/aurora/handlers/SubscribeHandlerTest.java:66]: Why are those tests commented.?

File: aurora/src/test/java/com/market/banica/aurora/observer/AuroraObserverTest.java [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+class AuroraObserverTest {
```
Comment (trayan-muchev) [inline aurora/src/test/java/com/market/banica/aurora/observer/AuroraObserverTest.java:19]: Conditions are missing in some of the test names.

File: aurora/src/test/java/com/market/banica/aurora/mapper/SubscribeMapperTest.java [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+    private void createFakeMarketReplyingServer() throws IOException {
+        grpcCleanup.register(InProcessServerBuilder
+                .forName(marketReceiverName)
+                .directExecutor()
+                .addService(this.fakeReceivingMarketService())
+                .build()
+                .start());
+    }
+
+    private void createFakeAuroraReplyingServer() throws IOException {
```
Comment (Lilyanov) [inline aurora/src/test/java/com/market/banica/aurora/mapper/SubscribeMapperTest.java:160]: This method is duplication with the one in RequestMapperTest. You can move all createFake...Server methods with the methods that create stubs in a parent class FakeStubsBaseTest or something similar and extend the current classes with it. Other option is to create util class that will be responsible for creating the fake servers and stubs

Comment (nikolanguen) [reply aurora/src/test/java/com/market/banica/aurora/mapper/SubscribeMapperTest.java]: I've created a new util package inside the test/java folder where I've implemented FakeStubsGenerator class to resolve the duplication issue.

Comment (Lilyanov) [reply aurora/src/test/java/com/market/banica/aurora/mapper/SubscribeMapperTest.java]: Can you move there the rest of the fake stubs and servers

**Review by Lilyanov — APPROVED** [review]

**Review by trayan-muchev — APPROVED** [review]

**Pull request merged**